src/lib/interactionManager/interaction.ts

```diff
@@ -292,21 +294,40 @@ export class Interaction {
     }
   }
 
-  public async storeCredential(toSave: SignedCredentialWithMetadata[]) {
+  public storeSelectedCredentials() {
```

Comment:
These two methods (`storeSelectedCredentials` and `storeCredentialMetadata`) are missing some error checking.
At least check that it is in fact a `CredentialOfferFlow`
And also check that the state has the necessary data (maybe the flow isn't through yet?)